src/TRestTrackLineAnalysisProcess.cxx

```diff
@@ -0,0 +1,252 @@
+/*************************************************************************
+ * This file is part of the REST software framework.                     *
+ *                                                                       *
+ * Copyright (C) 2016 GIFNA/TREX (University of Zaragoza)                *
+ * For more information see http://gifna.unizar.es/trex                  *
+ *                                                                       *
+ * REST is free software: you can redistribute it and/or modify          *
+ * it under the terms of the GNU General Public License as published by  *
+ * the Free Software Foundation, either version 3 of the License, or     *
+ * (at your option) any later version.                                   *
+ *                                                                       *
+ * REST is distributed in the hope that it will be useful,               *
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of        *
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the          *
+ * GNU General Public License for more details.                          *
+ *                                                                       *
+ * You should have a copy of the GNU General Public License along with   *
+ * REST in $REST_PATH/LICENSE.                                           *
+ * If not, see http://www.gnu.org/licenses/.                             *
+ * For the list of contributors see $REST_PATH/CREDITS.                  *
+ *************************************************************************/
+
+//////////////////////////////////////////////////////////////////////////
+/// The TRestTrackLineAnalysis process is meant to analyze lienar tracks,
+/// such as alpha tracks. It should retreive the information after performing
+/// TRestTrackLinearization process, for the time being only works when
```

Comment:
What it happens if `TRestTrackLinearizationProcess` was not used earlier?

Reply:
In fact, is not required to use `TRestTrackLinearizationProcess` before. It should work as soon as the reduced hits are ordered. I am going to change the description.

Reply:
Ok. It is this process doing a linear fit to the hits? In that case the hit order would also be not relevant.

Reply:
The hit order is required in the sense that it should follow the minimum path, this is guaranteed after performing `TRestTrackLinearizationProcess` . It can be also reached using `TRestTrackPathMinimizationProcess` after a `TRestTrackReductionProcess`.

However, at this stage the hits ordering doesn't require to have any preferred direction e.g. a hit ordering of `1->2->3->4->5` should give same solution as an inverted ordering `5->4->3->2->1`

Reply:
I see, we should probably introduce a mechanism that allows us to do something like `ProcessRequired("TRestTrackPathMinimizationProcess")` so that if the process does not exist it will be returning an error.

Another potential issue, perhaps @nkx111 sees an easy way

Reply:
I created an issue about it rest-for-physics/framework#182